java/arcs/core/analysis/Deduction.kt

```diff
@@ -0,0 +1,161 @@
+package arcs.core.analysis
+
+import arcs.core.data.expression.Expression
+
+/** Field [Identifier]. Lists of [Identifier]s imply an AccessPath.*/
+typealias Identifier = String
+
+/**
+ * Result of the [ExpressionClaimDeducer] on Paxel [Expression]s.
+ *
+ * [Deduction]s recursively associate identifiers to field paths. These relationships can be
+ * directly translated into [Claim]s and [AccessPath]s.
+ */
+sealed class Deduction {
+  /** Returns true if the [Deduction] collection has no members. */
+  open fun isEmpty(): Boolean = true
+
+  /** Returns true if the [Deduction] collection has some members. */
+  fun isNotEmpty(): Boolean = !isEmpty()
+
+  /** Unwrap [Deduction] object to get an underlying [Path]. */
+  abstract fun getPath(): Path
+
+  /** Substitute aliases in all paths. */
+  abstract fun substitute(aliases: Scope): Deduction
+
+  /** Union of two [Deduction] collections. */
+  abstract operator fun plus(other: Deduction): Deduction
+
+  /** A representation of a field path in a Paxel [Expression]. */
+  data class Path(val path: List<Identifier> = emptyList()) : Deduction() {
```

Comment:
If you `Path` is a representation of a simple field lookup, then maybe it's the same thing as `Equal` ?

Reply:
Hmm... `Path` is not quite a field lookup. `Derive` also composes a path, for example.